### PR TITLE
Adds an Indebted negative quirk to the game.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This Pull Request brings a new negative quirk to the list of negative quirks: Indebted. As the title suggests, you start the round with a pretty high debt of roughly 15.000 credits (give or take up to 1.250 credits), and everytime the account balance would increment, 75% of what's earned will be pushed toward solving it instead.

Unlike other quirks, it is hidding from medical HUDs and the health analyzers, because I frankly believe it's quite a stretch putting it on the same plane as other quirks which are of a more "psycho-physical" nature. However, examining the medical records will return a remark on how the quirk holder has had some difficulties paying the checkup bill, hinting it greatly. Examining their ID accurately will also report the entity of the debt.

Ok, the PR is now ready. It fancies an achievement and a pin too, for those who actually extinguish the debt.

## Why It's Good For The Game
I think it's cool to have a quirk that affects players from an angle different than most of the others. It doesn't affect the mob directly, brings no positive or negative moodlet, gives items, but it does cripple what they can do with their own bank account. Sure, it can be circumverted by "borrowing" another player mob's ID card, but frankly it has a mild negative value on par with "Family Heirloom" so it's not a big deal.

## Changelog

:cl:
add: Added an Indebted negative quirk to the game, which gives the holder's bank account a debt averaging at 15000 credits (with a variation of 1250 cr) and forces 75% of all earnings towards solving it. A little prize awaits those who actually extinguish it.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
could use something from Resettear for the cheevo name, but probably not the Capitalism, Ho! quote 🤔

**Comment by the-orange-cow** [comment]
Is there a way to manually send in funds to pay off the debt?

**Comment by Ghommie** [comment]
> Is there a way to manually send in funds to pay off the debt?

Not yet, but I'll make sure sure to add one.

> Edit: also, does this affect round start funds?

No, the initial 5 paychecks worth of funds are deposited before the quirk is added. I suppose it's ok to keep it that way, since it's going to be just a fraction of the whole debt whatsoever.

**Comment by Mickyan** [comment]
This is wildly out of scope but what would make this quirk so much more fun is if instead of getting a fixed number, your debt increases a bit every round but the amount you owe is saved through persistence (with a cap probably)

**Comment by AtomTheProphet** [comment]
> This is wildly out of scope but what would make this quirk so much more fun is if instead of getting a fixed number, your debt increases a bit every round but the amount you owe is saved through persistence (with a cap probably)

This would be _very_ nice.

**Comment by VladinXXV** [comment]
I think it'd be funny to just make the achievement name "Outdebted". In-debted => out-debted.

**Comment by EricZilla** [comment]
I think having a way to pay it off other than passively will make this just free points for certain jobs, but maybe you could have it also deduct from bounties as a method of actively paying it off? Like it takes 25% of bounties too

File: code/datums/quirks/negative_quirks/negative_quirks.dm [hunk 1]
```diff
@@ -1418,3 +1418,39 @@
...
+/datum/quirk/indebted/proc/on_debt_paid(datum/bank_account/source)
+	SIGNAL_HANDLER
+	if(source.account_debt)
+		return
+	UnregisterSignal(source, COMSIG_BANK_ACCOUNT_DEBT_PAID)
+	///The debt was extinguished while the quirk holder was logged out, so let's kindly award it once they come back.
+	if(!quirk_holder.client)
+		RegisterSignal(quirk_holder, COMSIG_MOB_LOGIN, PROC_REF(award_on_login))
+	else
+		quirk_holder.client.give_award(/datum/award/achievement/misc/debt_extinguished, quirk_holder)
```
Comment (san7890) [inline code/datums/quirks/negative_quirks/negative_quirks.dm:1451]: maybe something about garnishment for the name? idk

**Comment by YakumoChen** [comment]
> I think having a way to pay it off other than passively will make this just free points for certain jobs, but maybe you could have it also deduct from bounties as a method of actively paying it off? Like it takes 25% of bounties too

`			. += span_warning("The account is currently indebted for [registered_account.account_debt] cr. [100*DEBT_COLLECTION_COEFF]% of all earnings will go towards extinguishing it.")`

It takes off from any money you receive, not just paycheques, given the PR title says 75% that means it takes 3/4ths of every bounty you do too.

**Comment by Sealed101** [comment]
magnificent

File: code/datums/records/manifest.dm [hunk 2]
```diff
@@ -141,9 +141,9 @@ GLOBAL_DATUM_INIT(manifest, /datum/manifest, new)
 		trim = assignment,
 		// Crew specific
 		lock_ref = REF(lockfile),
-		major_disabilities = person.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY),
+		major_disabilities = person.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY, TRUE),
```
Comment (Jacquerel) [inline code/datums/records/manifest.dm:144]: Name these arguments please

File: code/game/objects/items/devices/scanners/health_analyzer.dm [hunk 3]
```diff
@@ -194,9 +194,9 @@
 				trauma_text += trauma_desc
 			render_list += "<span class='alert ml-1'>Cerebral traumas detected: subject appears to be suffering from [english_list(trauma_text)].</span>\n"
 		if(carbontarget.quirks.len)
-			render_list += "<span class='info ml-1'>Subject Major Disabilities: [carbontarget.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY)].</span>\n"
+			render_list += "<span class='info ml-1'>Subject Major Disabilities: [carbontarget.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY, TRUE)].</span>\n"
```
Comment (Jacquerel) [inline code/game/objects/items/devices/scanners/health_analyzer.dm:197]: These too

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -203,7 +203,7 @@
 				return
 
 			if(href_list["quirk"])
-				var/quirkstring = get_quirk_string(TRUE, CAT_QUIRK_ALL)
+				var/quirkstring = get_quirk_string(TRUE, CAT_QUIRK_ALL, TRUE)
```
Comment (Jacquerel) [inline code/modules/mob/living/carbon/human/human.dm:206]: And here

**Review by Jacquerel — CHANGES_REQUESTED** [review]
One small change and it looks good to go

File: code/datums/records/manifest.dm (hunk 2 above)
Comment (Ghommie) [reply code/datums/records/manifest.dm:144]:
```suggestion
		major_disabilities = person.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY, from_scan = TRUE),
```

File: code/game/objects/items/devices/scanners/health_analyzer.dm (hunk 3 above)
Comment (Ghommie) [reply code/game/objects/items/devices/scanners/health_analyzer.dm:197]:
```suggestion
			render_list += "<span class='info ml-1'>Subject Major Disabilities: [carbontarget.get_quirk_string(FALSE, CAT_QUIRK_MAJOR_DISABILITY, from_scan = TRUE)].</span>\n"
```

**Comment by VladinXXV** [comment]
image
done

**Comment by Sealed101** [comment]
what a sweatlord smh

**Pull request merged**